src/Analyzers/MSTest.Analyzers/DynamicDataShouldBeValidAnalyzer.cs

```diff
@@ -275,13 +275,17 @@ private static void AnalyzeDataSource(SymbolAnalysisContext context, AttributeDa
 
         // Validate member return type.
         ITypeSymbol? memberTypeSymbol = member.GetMemberType();
-        if (memberTypeSymbol is INamedTypeSymbol memberNamedType
-            && (!SymbolEqualityComparer.Default.Equals(memberNamedType.ConstructedFrom, ienumerableTypeSymbol)
-                || memberNamedType.TypeArguments.Length != 1))
+        if (memberTypeSymbol is not INamedTypeSymbol memberNamedType)
         {
-            context.ReportDiagnostic(attributeSyntax.CreateDiagnostic(MemberTypeRule, declaringType.Name, memberName));
             return;
```

Comment:
Looks like a false negative to me. Let's say the method returns a pointer type. We will return right away because that's not an `INamedTypeSymbol`